includes/polling/unix-agent/packages.inc.php

```diff
@@ -5,158 +5,76 @@
+        $package->status = 0;
+
+        return $package;
+    })->keyBy->getCompositeKey();
 
     foreach (explode("\n", $agent_data['rpm']) as $package) {
         [$name, $version, $build, $arch, $size] = explode(' ', $package);
-        $pkgs[$manager][$name][$arch][$version][$build]['manager'] = $manager;
-        $pkgs[$manager][$name][$arch][$version][$build]['name'] = $name;
-        $pkgs[$manager][$name][$arch][$version][$build]['arch'] = $arch;
-        $pkgs[$manager][$name][$arch][$version][$build]['version'] = $version;
-        $pkgs[$manager][$name][$arch][$version][$build]['build'] = $build;
-        $pkgs[$manager][$name][$arch][$version][$build]['size'] = $size;
-        $pkgs[$manager][$name][$arch][$version][$build]['status'] = '1';
-        $text = $manager . '-' . $name . '-' . $arch . '-' . $version . '-' . $build;
-        $pkgs_id[] = $pkgs[$manager][$name][$arch][$version][$build];
+        $package = new \App\Models\Package([
+            'manager' => $manager,
+            'name' => $name,
+            'arch' => $arch,
+            'version' => $version,
+            'build' => $build,
+            'size' => $size,
+            'status' => 1,
+        ]);
+        $package_key = $package->getCompositeKey();
+        if ($existing_package = $packages->get($package_key)) {
+            $existing_package->fill($package->attributesToArray());
+        } else {
+            $packages->put($package_key, $package);
+        }
     }
 }//end if
 
 // DPKG
 if (! empty($agent_data['dpkg'])) {
-    echo "\nDEB Packages: ";
+    echo "\nDEB Packages: \n";
     // Build array of existing packages
     $manager = 'deb';
 
-    $pkgs_db_db = dbFetchRows('SELECT * FROM `packages` WHERE `device_id` = ?', [$device['device_id']]);
-    foreach ($pkgs_db_db as $pkg_db) {
-        $pkgs_db[$pkg_db['manager']][$pkg_db['name']][$pkg_db['arch']][$pkg_db['version']][$pkg_db['build']]['id'] = $pkg_db['pkg_id'];
-        $pkgs_db[$pkg_db['manager']][$pkg_db['name']][$pkg_db['arch']][$pkg_db['version']][$pkg_db['build']]['status'] = $pkg_db['status'];
-        $pkgs_db[$pkg_db['manager']][$pkg_db['name']][$pkg_db['arch']][$pkg_db['version']][$pkg_db['build']]['size'] = $pkg_db['size'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['text'] = $pkg_db['manager'] . '-' . $pkg_db['name'] . '-' . $pkg_db['arch'] . '-' . $pkg_db['version'] . '-' . $pkg_db['build'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['manager'] = $pkg_db['manager'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['name'] = $pkg_db['name'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['arch'] = $pkg_db['arch'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['version'] = $pkg_db['version'];
-        $pkgs_db_id[$pkg_db['pkg_id']]['build'] = $pkg_db['build'];
-    }
+    /** @var \Illuminate\Support\Collection $packages */
+    $packages = DeviceCache::getPrimary()->packages->map(function (\App\Models\Package $package) {
```

Comment:
It can be further simplified by removing these duplicated functions

Reply:
Yes, but I'm not going to convert it to class base code.  And it isn't much of an improvement with functional code.